packages/opencensus-core/src/trace/model/tracer.ts

```diff
@@ -30,7 +30,7 @@ import * as types from './types';
 /**
  * This class represent a tracer.
  */
-export class Tracer implements types.Tracer {
+export class TracerAgent implements types.Tracer {
```

Comment:
I much prefer `Tracer` as the class name, especially because this class is publicly exposed. In other words, I would rather the interface name change rather than the class. Or, I believe that if there is a 1:1 association between an interface and an implementation, that there not be an interface at all -- and this is one of those situations.

Reply:
We can identify a general convention at @opencensus/core, there are always interfaces for exported classes.  I personly like the idea of a Tracer interface here. All instrumentations rely on that interface, I think that is a good approach to rely on interfaces rather than concrete implementations.  If we think of interfaces as 'contracts' between the system and its clients, the way it is been used by the core package is very natural.

Reply:
> I think that is a good approach to rely on interfaces rather than concrete implementations

That is fair. Then I would suggest changing the interface name. For instrumentation, we don't use the entire `Tracer` interface; as an example, we actually don't want instrumentation to start or stop the tracer, access the sampler, or register span event listeners (AFAIK). Maybe we can define an interface that specifies the type of the object passed to instrumentation (like `SpanCreator`), and a different interface for the type of object that is being managed by the `Tracing` instance (like `SpanEventManager`). Then we would just have `class Tracer implements SpanCreator, SpanEventManager`. For anything that doesn't fit in either of these categories, we should ask whether it needs to be publicly exposed at all. (The names are just suggestions, I don't have strong feelings about them.)

I don't want to force my opinion on this one, so if you disagree that's fine; I would still prefer that a different name be given to the class. Maybe `BaseTracer` (it's not entirely accurate, but I can't think of anything better than that).

Reply:
> .. Maybe we can define an interface that specifies the type of the object passed to instrumentation (like SpanCreator), and a different interface for the type of object that is being managed by the Tracing instance (like SpanEventManager).  and a different interface for the type of object that is being managed by the Tracing instance (like SpanEventManager). Then we would just have class Tracer implements SpanCreator, SpanEventManager.

I understand the reasons, but I'm really not sure if creating this new set of interfaces could bring any benefit to the code, on the contrary, maybe add complexity. The "tracer" name representing the manager/controller class of Spans is a common name in the domain space, it is used by Opencensus and others Trace solutions.

Regarding the name of the class, I've chosen 'CoreTracer' over 'BaseTracer', cause the word Base feels like we have to build something over it.

Reply:
Thanks. `CoreTracer` is fine.